src/vs/workbench/contrib/notebook/browser/view/renderers/backLayerWebView.ts

```diff
@@ -1167,21 +1158,35 @@ var requirejs = (function() {
 			useRootAuthority: true
 		}, undefined);
 
-		let resolveFunc: () => void;
-		this._loaded = new Promise<void>((resolve, reject) => {
-			resolveFunc = resolve;
-		});
+		webview.html = content;
+		return webview;
+	}
 
-		const dispose = webview.onMessage((message) => {
-			const data: FromWebviewMessage = message.message;
-			if (data.__vscode_notebook_message && data.type === 'initialized') {
-				resolveFunc();
-				dispose.dispose();
+	private initializeWebViewState() {
+		const renderers = new Set<INotebookRendererInfo>();
+		for (const inset of this.insetMapping.values()) {
+			if (inset.renderer) {
+				renderers.add(inset.renderer);
 			}
-		});
+		}
 
-		webview.html = content;
-		return webview;
+		this._preloadsCache.clear();
+		if (this._currentKernel) {
+			this._updatePreloadsFromKernel(this._currentKernel);
+		}
+		this.updateRendererPreloads(renderers);
+
+		for (const [output, inset] of this.insetMapping.entries()) {
+			this._sendMessageToWebview({ ...inset.cachedCreation, initiallyHidden: this.hiddenInsetMapping.has(output) });
+		}
+
+		const mdCells = [...this.markdownPreviewMapping.values()];
+		this.markdownPreviewMapping.clear();
+		for (const cell of mdCells) {
+			if (cell.visible) {
+				this.createMarkdownPreview(cell);
```

Comment:
Can you use `initializeMarkdownPreview`  instead? This will create the cells one a time, which is more likely to cause flickering